### PR TITLE
Optional files don't cause assignment to be marked as partial

### PR DESCRIPTION
**Pull request by maxnz**
Missing optional files were causing assignments to be marked as partially complete even if all required files were present. This removes optional files from the analysis.

### CODE REVIEW

File: cs251tk/specs/util.py [hunk 1]
```diff
@@ -4,7 +4,15 @@
 
 def get_filenames(spec):
     """returns the list of files from an assignment spec"""
-    return [file['filename'] for file in spec['files']]
+    files = [file['filename'] for file in spec['files']]
```
Comment (hawkrives) [inline cs251tk/specs/util.py]: `return [file['filename'] for file in spec['files'] if not file[options].get('optional', False)]`

Comment (maxnz) [reply cs251tk/specs/util.py]: The problem with that is that it doesn't catch KeyErrors, which are almost guaranteed for required files

Comment (maxnz) [reply cs251tk/specs/util.py]: Nevermind, that does work. `options` should be in quotes, that's where the error was coming from.

**Comment by maxnz** [comment]
I think this will be my last update before the next release. Any chance that could be today?

**Comment by maxnz** [comment]
More specifically, could we have a release before the SD lab today at 3?

**Comment by hawkrives** [comment]
Should be out in a few minutes as v2.7.0: https://github.com/StoDevX/cs251-toolkit/releases/tag/v2.7.0

**Comment by maxnz** [comment]
Thanks!

**Pull request merged**